inc/providers/class-wp-sitemaps-users.php

```diff
@@ -81,25 +105,41 @@ public function get_url_list( $page_num, $object_subtype = '' ) {
 	 *                               provider class. Default empty.
 	 * @return int Total page count.
 	 */
-	public function max_num_pages( $object_subtype = '' ) {
-		$query = $this->get_public_post_authors_query();
+	public function get_max_num_pages( $object_subtype = '' ) {
+		/**
+		 * Filters the max number of pages before it is generated.
+		 *
+		 * Passing a non-null value will effectively short-circuit the generation,
+		 * returning that value instead.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param int $max_num_pages The maximum number of pages. Default null.
+		 */
+		$max_num_pages = apply_filters( 'wp_sitemaps_users_pre_max_num_pages', null );
+
+		if ( null !== $max_num_pages ) {
+			return $max_num_pages;
+		}
+
+		$args = $this->get_users_query_args();
+		$args['paged'] = 1;
```

Comment:
Is this actually needed here?

Reply:
I just copied this but... seems like it, yeah.